R/coercion.r

```diff
@@ -407,10 +407,8 @@ setMethod("as.period", signature(x = "Interval"), function(x, unit = NULL, ...)
   switch(unit,
     year = .int_to_period(x),
     month = {
-      pers <- .int_to_period(x)
-      month(pers) <- month(pers) + year(pers) * 12
-      year(pers) <- 0
-      pers
+      x <- .int_to_period(x)
+      as.period(x, unit = unit)
```

Comment:
Standardized to just call `as.period()` to go through the Period method, since at this point the code paths are exactly the same. Hopefully this makes them less likely to get out of sync going forward